src/strategies/singleSlotProof.ts

```diff
@@ -8,15 +9,46 @@ import type {
   VanillaProposeMessage,
   VanillaVoteMessage
 } from '../types';
+import { getStorageVarAddress } from '../utils/encoding';
 
+const strategyParamsStore = 'Voting_voting_strategy_params_store';
 const fossilFactRegistryAddress =
   '0x363108ac1521a47b4f7d82f8ba868199bc1535216bbedfc1b071ae93cc406fd';
 
+async function fetchStrategyParams(
+  address: string,
+  envelope: Envelope<VanillaProposeMessage | VanillaVoteMessage>
+): Promise<string[]> {
+  const lengthAddress = getStorageVarAddress(strategyParamsStore, address, '0x0');
+  const length = parseInt(
+    (await defaultProvider.getStorageAt(envelope.data.message.space, lengthAddress)) as string,
+    16
+  );
+
+  return Promise.all(
+    [...Array(length)].map(async (_, i) => {
```

Comment:
How many requests this trigger to get SSP params? 2?

Reply:
1 to get length first, then 2 to fetch the params itself.

If we are sure that there will be always 2 params we could skip first request.